appsec/config_test.go

```diff
@@ -0,0 +1,243 @@
+				expCfg := defaultConfig
+				expCfg.KeyRegex = "test"
+				t.Setenv(envObfuscatorKey, "test")
+				cfg := NewObfuscatorConfig()
+				require.Equal(t, expCfg, cfg)
+			})
+			t.Run("env-var-empty", func(t *testing.T) {
+				expCfg := defaultConfig
+				expCfg.KeyRegex = ""
+				t.Setenv(envObfuscatorKey, "")
+				cfg := NewObfuscatorConfig()
+				require.Equal(t, expCfg, cfg)
+			})
+			t.Run("compile-error", func(t *testing.T) {
+				t.Setenv(envObfuscatorKey, "+")
+				cfg := NewObfuscatorConfig()
+				require.Equal(t, defaultConfig, cfg)
+			})
+		})
+
+		t.Run("value-regexp", func(t *testing.T) {
+			t.Run("env-var-normal", func(t *testing.T) {
+				expCfg := defaultConfig
+				expCfg.ValueRegex = "test"
+				t.Setenv(envObfuscatorValue, "test")
+				cfg := NewObfuscatorConfig()
+				require.Equal(t, expCfg, cfg)
+			})
+			t.Run("env-var-empty", func(t *testing.T) {
+				expCfg := defaultConfig
+				expCfg.ValueRegex = ""
+				t.Setenv(envObfuscatorValue, "")
+				cfg := NewObfuscatorConfig()
+				require.Equal(t, expCfg, cfg)
+			})
+			t.Run("compile-error", func(t *testing.T) {
+				t.Setenv(envObfuscatorValue, "+")
+				cfg := NewObfuscatorConfig()
+				require.Equal(t, defaultConfig, cfg)
+			})
+		})
+	})
+}
+
+func TestTraceRateLimit(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		env      string
+		expected uint
+	}{
+		{
+			name:     "parsable",
+			env:      "1234567890",
+			expected: 1234567890,
+		},
+		{
+			name:     "not-parsable",
+			env:      "not a uint",
+			expected: defaultTraceRate,
+		},
+		{
+			name:     "negative",
+			env:      "-1",
+			expected: defaultTraceRate,
+		},
+		{
+			name:     "zero",
+			env:      "0",
+			expected: defaultTraceRate,
+		},
+		{
+			name:     "empty-string",
+			env:      "",
+			expected: defaultTraceRate,
+		},
+	} {
+		t.Run("trace-rate-limit/"+tc.name, func(t *testing.T) {
+			t.Setenv(envTraceRateLimit, tc.env)
+			require.Equal(t, tc.expected, RateLimitFromEnv())
+		})
+	}
+}
+
+func TestWAFTimeout(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		env      string
+		expected time.Duration
+	}{
+		{
+			name:     "parsable",
+			env:      "5s",
+			expected: 5 * time.Second,
+		},
+		{
+			name:     "parsable-default-microsecond",
+			env:      "1",
+			expected: 1 * time.Microsecond,
+		},
+		{
+			name:     "not-parsable",
+			env:      "not a duration string",
+			expected: defaultWAFTimeout,
+		},
+		{
+			name:     "negative",
+			env:      "-1s",
+			expected: defaultWAFTimeout,
+		},
+		{
+			name:     "zero",
+			env:      "0",
+			expected: defaultWAFTimeout,
+		},
+		{
+			name:     "empty-string",
+			env:      "",
+			expected: defaultWAFTimeout,
+		},
+	} {
+		t.Run("waf-timeout/"+tc.name, func(t *testing.T) {
+			t.Setenv(envWafTimeout, tc.env)
+			require.Equal(t, tc.expected, WAFTimeoutFromEnv())
+		})
+
+	}
+}
+
+func TestRules(t *testing.T) {
+	t.Run("rules", func(t *testing.T) {
```

Comment:
Same as above - there is a single block there so it seems redundant...

Reply:
af96bef40d5e6